packages/typespec-ts/src/modular/buildClassicalClient.ts

```diff
@@ -207,6 +210,8 @@ function buildClientOperationGroups(
       operationGroup.namespaceHierarchies[0] ?? operationGroup.propertyName,
       NameType.Property
     );
+    // TODO: remove this logic once client-level parameter design is finalized
+    const hasSubIdPromoted = shouldPromoteSubId(dpgContext, operationGroup);
```

Comment:
lets use the full name for clarity `hasSubscriptionIdPromoted` and `shouldPromoteSubscriptionId`

Reply:
done